go.mod

```diff
@@ -1,16 +1,280 @@
-module github.com/a2aproject/a2a
+module github.com/a2aproject/a2a-go
 
 go 1.24.4
 
+tool (
+	github.com/bufbuild/buf/cmd/buf
+	github.com/golangci/golangci-lint/cmd/golangci-lint
+	google.golang.org/grpc/cmd/protoc-gen-go-grpc
+	google.golang.org/protobuf/cmd/protoc-gen-go
+)
+
 require (
 	google.golang.org/genproto/googleapis/api v0.0.0-20250715232539-7130f93afb79
 	google.golang.org/grpc v1.73.0
 	google.golang.org/protobuf v1.36.6
 )
 
 require (
+	4d63.com/gocheckcompilerdirectives v1.3.0 // indirect
```

Comment:
This clearly makes the go.mod dependency graph large, and that is why I have been against the tool feature. :-(
 
We should use `go install <tool>@version`, instead of polluting the dependencies for most tools unless there is a clear reason that the main project needs to share the specific version aligned with the tools.

Reply:
The problem of the tools dependency I see as a former maintainer of the tools from the go team (e.g. gopls, go command, etc) is that, 1) the dependencies used to build a tool are not the tool author used and tested, and 2) the tools' dependencies are taken into account when building the module version graph for the project and influence project's dependencies. For example, some tools want to use the latest `go` to work correctly. Adding such tool as a tool dependency will update the go version in go.mod. For some tools, that may be useful. I am not sure if this is what we want for this project yet.